packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts

```diff
@@ -43,12 +50,16 @@ export interface LambdaRestApiProps extends RestApiProps {
  */
 export class LambdaRestApi extends RestApi {
   constructor(scope: Construct, id: string, props: LambdaRestApiProps) {
-    if ((props.options && props.options.defaultIntegration) || props.defaultIntegration) {
+    if (props.options?.defaultIntegration || props.defaultIntegration) {
       throw new Error('Cannot specify "defaultIntegration" since Lambda integration is automatically defined');
     }
+    // Forbid setting `integrationOptions.proxy = false` unless `proxy = false`
+    if (props.integrationOptions?.proxy !== undefined && props.integrationOptions?.proxy !== (props?.proxy ?? true)) {
```

Comment:
Please simplify this `if` as well, it's very complicated right now. 

I think it should be broken apart like this:

```
     if (props.integrationOptions?.proxy !== undefined) { 
       if (props.integrationOptions.proxy !== (props?.proxy ?? true)) { ...
         // ...
       }
     }
```

and I think the `props?.proxy ?? true` should be extracted into a variable with a name that describes what that part of the expression represents.